elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -635,13 +635,38 @@ public void removeRelation(String fieldName, PersistentResource removeResource)
         }
     }
 
+    /**
+     * Checks whether the new entity is already a part of the relationship
+     * @param fieldName which relation link
+     * @param toAdd the new relation
+     * @return boolean representing the result of the check
+     *
+     * This method does not handle the case where the toAdd entity is newly created,
+     * since newly created entities have null id there is no easy way to check for identity
+     */
+    public boolean relationshipAlreadyExists(String fieldName, PersistentResource toAdd) {
+        Object relation = this.getValueUnchecked(fieldName);
+        String toAddId = toAdd.getId();
+        if (toAddId == null) {
+            return false;
+        }
+        if (relation instanceof Collection) {
+            return ((Collection) relation).stream().anyMatch((obj -> dictionary.getId(obj).equals(toAddId)));
```

Comment:
Let's swap the operands of the equals here since we only checked for null on toAddId.